gems/smithy-client/spec/smithy-client/plugins/resolve_auth_spec.rb

```diff
@@ -87,22 +83,22 @@ def resolve(_)
         it 'raises an error when auth scheme is not enabled' do
           shapes['smithy.ruby.tests#SampleClient']['traits']['smithy.api#httpApiKeyAuth'] = {}
           client_class.remove_plugin(HttpApiKeyAuth)
-          expect { client.operation }.to raise_error(/was not enabled/)
+          expect { client.operation }.to raise_error(/was not enabled for this request/)
         end
 
         it 'raises an error when identity resolver is not configured' do
           shapes['smithy.ruby.tests#SampleClient']['traits']['smithy.api#httpApiKeyAuth'] = {}
           client_class.add_plugin(HttpApiKeyAuth)
-          client = client_class.new(stub_responses: true, http_api_key_provider: nil)
-          expect { client.operation }.to raise_error(/did not have an identity resolver configured/)
+          client = client_class.new(stub_responses: true, api_key_provider: nil)
+          expect { client.operation }.to raise_error(/did not have an identity provider configured/)
         end
 
         it 'raises an error when identity resolver fails to resolve identity' do
           shapes['smithy.ruby.tests#SampleClient']['traits']['smithy.api#httpApiKeyAuth'] = {}
           client_class.add_plugin(HttpApiKeyAuth)
-          provider = HttpApiKeyProvider.new('stubbed_key')
+          provider = ApiKeyProvider.new(key: 'stubbed_key')
           expect(provider).to receive(:identity).and_return(nil)
```

Comment:
We can improve this test. Just pass an empty instance of the provider. Set? Should return false on its own.